### ISSUE TITLE
Training loop missing

### ISSUE THREAD

**Issue by ujwaln15** [Chilie/Deblur_MCEM#1]
Please provide the training loop with loss calculation, backpropagation and scheduler incorporated.
